### ISSUE TITLE
Offer to save machine state on exit

### ISSUE THREAD

**Issue by mborgerson** [xemu-project/xemu#1375]
### Feature Request

Once #782 lands, it would be nice to offer users the option to save a snapshot when they close xemu, and to load it back up again when they open xemu.

**Comment by Spidy123222** [comment]
Will there be a way to turn off that prompt. Also would it be able to be selected if yes or no on gamepad natively like the gamepadui?

**Comment by davidhedlund** [comment]
From https://github.com/xemu-project/xemu/issues/2710 that just was closed as a duplicate of this issue:

## Example: [PCSX2](https://github.com/PCSX2/pcsx2) menu UI

### Menu bar
* **Shut Down** -- Prompts "Are you sure you want to shut down the virtual machine?" Yes (button) No (button) "Save state for Resume" (checkbox)
* **Shut Down Without Saving**
* **Exit** -- Displays the same confirmation prompt as Shut Down

### Pause menu
* **Close Game**
  - **Exit and Save State**
  - **Exit Without Saving**
